### ISSUE TITLE
Wrong output shapes in Whisper encoder

### ISSUE THREAD

**Issue by MahmoudAshraf97** [NVIDIA/TensorRT-LLM#2689]
### System Info

TRT-LLM version: 0.17.0.dev2024121700
GPU: H100, L40s

### Who can help?

@byshiue 

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
import json
import os

import tensorrt_llm.bindings.executor as trtllm
import torch

from whisper.tokenizer import get_tokenizer
from math import ceil

engine_path = "TRT_LLM/whisper_large-v3-turbo_int8"

num_beams = 1
executor = trtllm.Executor(
    encoder_model_path=os.path.join(engine_path, "encoder"),
    decoder_model_path=os.path.join(engine_path, "decoder"),
    model_type=trtllm.ModelType.ENCODER_DECODER,
    executor_config=trtllm.ExecutorConfig(
        num_beams,
        max_batch_size=96,
        kv_cache_config=trtllm.KvCacheConfig(
            free_gpu_memory_fraction=0.5, cross_kv_cache_fraction=0.5
        ),
        parallel_config=trtllm.ParallelConfig(device_ids=[1]),
    ),
)
with open(os.path.join(engine_path, "encoder", "config.json"), "r") as f:
    encoder_config = json.load(f)
n_vocab = encoder_config["pretrained_config"]["vocab_size"]
n_mels = encoder_config["pretrained_config"]["n_mels"]
is_multilingual = n_vocab >= 51865

num_languages = n_vocab - 51765 - int(is_multilingual)
tokenizer = get_tokenizer(multilingual=is_multilingual, num_languages=num_languages)

prompt = tokenizer.encode(
    "<|startoftranscript|><|en|><|transcribe|><|notimestamps|>",
    allowed_special=set(tokenizer.special_tokens),
)

features = [
    torch.randn(128, 659).half(),
    torch.randn(128, 483).half(),
]
requests = [
    trtllm.Request(
        input_token_ids=prompt,
        max_tokens=210,
        encoder_input_features=feature.T.half().contiguous(),  # mel features
        encoder_output_length=ceil(feature.shape[-1] / 2),
        end_id=tokenizer.eot,
        pad_id=tokenizer.eot,
        sampling_config=trtllm.SamplingConfig(beam_width=1),
    )
    for feature in features
]

request_ids = executor.enqueue_requests(requests)

results = executor.await_responses(request_ids)
```
```
[TensorRT-LLM][ERROR] IExecutionContext::inferShapes: Error Code 7: Internal Error (WhisperEncoder/__add___L322/elementwise_binary_L2890/ELEMENTWISE_SUM_0: dimensions not compatible for elementwise. Broadcast has incompatible dimensions: 572 != 571 && 572 != 1 && 571 != 1. Instruction: CHECK_BROADCAST 572 571.)
[TensorRT-LLM][ERROR] Encountered an error in forwardAsync function: Invalid input shape (/home/jenkins/agent/workspace/LLM/main/L0_Test-x86_64/tensorrt_llm/cpp/tensorrt_llm/runtime/tllmRuntime.cpp:574)
```

Each request alone works fine but both running together throws the error, the correct summed sequence length for both input is 572 while TRT generated 571
the correct output shapes can be verified using this simple module that simulates whisper large v3 conv modules:
```
class MyConv(torch.nn.Module):
    def __init__(self):
        super(MyConv, self).__init__()
        self.conv1 = torch.nn.Conv1d(128, 1280, kernel_size=3, stride=1, padding=1)
        self.conv2 = torch.nn.Conv1d(1280, 1280, kernel_size=3, stride=2, padding=1)

    def forward(self, x):
        return self.conv2(self.conv1(x))

my_conv = MyConv()
for i in range(101,3000,2):
    with torch.no_grad():
        x = torch.randn(1, 128, i)
        y = my_conv(x)
        assert y.shape == (1, 1280, ceil(i/2))

### Expected behavior

.

### actual behavior

.

### additional notes

.

**Comment by yuekaizhang** [comment]
@MahmoudAshraf97  It seems you're using the Whisper IFB feature. For the official Whisper checkpoint, you need to pad the WAV file to 30 seconds. For fine-tuned Whisper, which can accept WAV files shorter than 30 seconds, you should pad it like this: https://github.com/NVIDIA/TensorRT-LLM/blob/main/examples/whisper/run.py#L533

**Comment by MahmoudAshraf97** [comment]
That's what I ended up doing at the end, but I thought that I should report the bug

**Comment by yuekaizhang** [comment]
> That's what I ended up doing at the end, but I thought that I should report the bug

Thanks. However, for IFB, we pack the requests. If we don't use padding, the head of the second request and the tail of the first request will be mixed by the convolution layer. I currently don't have a better solution in mind.
